rskj-core/src/main/java/co/rsk/net/SnapshotProcessor.java

```diff
@@ -219,11 +222,12 @@ public void run() {
 
     void processSnapStatusRequestInternal(Peer sender, SnapStatusRequestMessage requestMessage) {
         long bestBlockNumber = blockchain.getBestBlock().getNumber();
-        long checkpointBlockNumber = bestBlockNumber - (bestBlockNumber % BLOCK_NUMBER_CHECKPOINT);
+        long checkpointBlockNumber = Math.max(0, bestBlockNumber - checkpointDistance);
```

Comment:
What is a `checkpointBlockNumber`? could you elaborate on what happened that we are doing this changes?